test/utils/utils.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+	"golang.org/x/oauth2/google"
+	alpha "google.golang.org/api/compute/v0.alpha"
+	beta "google.golang.org/api/compute/v0.beta"
+	"google.golang.org/api/compute/v1"
+)
+
+const currentNodeCountRE = `currentNodeCount: (.*)\n`
+
+var (
+	errNotMatchFound       = errors.New("no match found for the given regex.")
+	errTypeAssersionFailed = errors.New("type assertion failed")
+	errValueNotMatch       = errors.New("value not match for the given field")
+)
+
+func newCloud(project string) (cloud.Cloud, error) {
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, compute.ComputeScope)
+	if err != nil {
+		return nil, err
+	}
+
+	alpha, err := alpha.New(client)
+	if err != nil {
+		return nil, err
+	}
+	beta, err := beta.New(client)
+	if err != nil {
+		return nil, err
+	}
+	ga, err := compute.New(client)
+	if err != nil {
+		return nil, err
+	}
+
+	svc := &cloud.Service{
+		GA:            ga,
+		Alpha:         alpha,
+		Beta:          beta,
+		ProjectRouter: &cloud.SingleProjectRouter{ID: project},
+		RateLimiter:   &cloud.NopRateLimiter{},
+	}
+
+	theCloud := cloud.NewGCE(svc)
+	return theCloud, nil
+}
+
+func verifyClusterInfo(input string, numOfNodes int) error {
```

Comment:
you should document the format of the input string that is coming in.

Reply:
Added in the comment.